### PR TITLE
- normalizes the string representation by wrapping around smaller units if required

### PR DESCRIPTION
**Pull request by baywet**
# Description

This PR adds a normalize function to wrap around smaller units when required.
I was wondering whether it should be exposed publicly so users can call it themselves should they want need to?
I was also wondering whether it should be called in the "ToDuration" method?
Thanks for reviewing it!

## Type of change

Please delete options that are not relevant.

- [ ] Documentation Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have run tests (go test) that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by baywet** [comment]
ping @cjlapao

**Comment by cjlapao** [comment]
> ping @cjlapao 

Didn't know anyone was using this library, there is a massive revamp of it in a branch so be aware if you are using this there will be a lot of broken changes coming in the next week or so

**Comment by baywet** [comment]
Thanks for sharing that information. Breaking changes are ok, we're in a preview state on our end as well. We'll upgrade/resolve whenever the newer version comes out. Unless you recommend an alternative?

**Comment by cjlapao** [comment]
> Thanks for sharing that information. Breaking changes are ok, we're in a preview state on our end as well. We'll upgrade/resolve whenever the newer version comes out. Unless you recommend an alternative?

No I created this because I couldn't find a lot of the things I need, the new one will bring a open Id kind of identity management for the API and a mongo database wrapper for easy creation of pipelines. Also one of your tests is failing

**Comment by baywet** [comment]
The only part we have a dependency on at this point [is the duration](https://github.com/microsoft/kiota/blob/2884fded60c2a6dc2ef07b0cd97bc703e2d20dbd/abstractions/go/serialization/iso_duration.go#L6) and we're not planning on taking a dependency on anything else. As a matter of fact, making the duration package independent would help people who need it adopt it.

Yep, I saw the test thing, looking into it.

**Comment by baywet** [comment]
Just pushed a new commit to address the unit test issue, let me know if you need me to fix anything else.

**Comment by baywet** [comment]
Also, after you're done reviewing the code, and before this gets merged, you might want to review those outstanding questions I had.
> I was wondering whether it should be exposed publicly so users can call it themselves should they want need to?
> I was also wondering whether it should be called in the "ToDuration" method?

**Comment by cjlapao** [comment]
> Also, after you're done reviewing the code, and before this gets merged, you might want to review those outstanding questions I had.
> 
> > I was wondering whether it should be exposed publicly so users can call it themselves should they want need to?
> > I was also wondering whether it should be called in the "ToDuration" method?

No Problem in exposing it publicly as it seems to be useful and yes I think we should use it in the ToDuration also

**Review by cjlapao — APPROVED** [review]

**Comment by baywet** [comment]
done, thanks for reviewing!
I do think we should also add support for months as described in #12
If you agree with the statements in there, I can get started on another PR.

**Comment by cjlapao** [comment]
> done, thanks for reviewing! I do think we should also add support for months as described in #12 If you agree with the statements in there, I can get started on another PR.

Yes that seems to be a great idea, if you have the time then please do

**Comment by baywet** [comment]
I'll probably do that as a separate PR so it doesn't block this one. You can go ahead and merge.

**Pull request merged**
